lib/Sema/CSSimplify.cpp

```diff
@@ -6000,20 +6129,24 @@ ConstraintSystem::simplifyCheckedCastConstraint(
     Type toKeyType, toValueType;
     std::tie(toKeyType, toValueType) = *isDictionaryType(toType);
 
-    if (simplifyCheckedCastConstraint(fromKeyType, toKeyType, subflags,
+    if (simplifyCheckedCastConstraint(fromKeyType, toKeyType,
+                                      subflags | TMF_ApplyingFix,
```

Comment:
Is `TMF_ApplyingFix` always appropriate here?

Reply:
Ah, I was thinking about it ... the only reason for this is to avoid recording the fix for the element types. What do you think about using a new locator path for that e.g. `KnownStdlibCollectionElement`? I think that may be a better approach to this.

Reply:
I just added this so let me know what you think, I think is reasonable(maybe could opt for other name) but just let me know, I can revert no problem =]